### ISSUE TITLE
Add a new gradebook functionality: Minimum score

### ISSUE THREAD

**Issue by jmarianolujan** [sakaiproject/sakai#1595]
Instructors are interested in having a new option in gradebook that allows them to set a minimum score. Its kind of a 'conditional release'.

Minimum score: It is the minimum score necessary to reach in the gradebook item to pass the course.
Specifications:

    Add a new gradebook item field: Minimum score
    If the student reaches less than the minimum score his course grade will be blank.

**Comment by jmarianolujan** [comment]
This is really a feature request, I tried adding a label to it 'enhacements' but I can't.

**Comment by steveswinsburg** [comment]
Could this be handled with the minimum score in the course grade? You could maybe use the Pass/Not Pass setting and set the applicable minimum grade that way? If not, how did you envisage it working?

**Comment by jonespm** [comment]
I think this sound complicated with a lot of possibilities and maybe not something I'd want to see tackled right now. This is the same description as the request at https://jira.sakaiproject.org/browse/GRBK-1429.

Would this be minimum score for the course or minimum score for any individual item? The way it's described in the picture on GRBK-1429, if they got below the minimum score even on a single item they don't get a course grade, that doesn't make sense to me. 

I also feel like many people would interpret this more like "this is the minimum points it's going to assign for completion". So instead of 0-100 possible points.(for 100 points) it's 60-100 possible points, no matter how badly a score it is. (Like of like a curved grade) 

It seems really confusing to me when it's locking down the determination of the final course grade based on some items.

**Comment by jmarianolujan** [comment]
Hi Steve and Matthew,
Thanks for your comments, let me clarify it a bit more.
As Matthew is saying, it is the same issue described in GBRK-1429 (that one was intended for GB2) and since we are planning on moving from GB2 to GBNG, we will be opening github issues for all the functionality missing in GBNG, to start the discussion and prioritize functionalty with others from the community: https://confluence.sakaiproject.org/display/SG2X/Moving+from+Gradebook2+to+GradebookNG

So, back to this issue:

This is a piece of functionality that was included in our old 'custom' grading system and that Instructors at our institution still ask for. It is intended only for individual grading items, it's NOT a minimum course score, its a minimum gradebook item score that students need to achieve in order to get a FINAL GRADE. Here is simple example:

MATHS - FINAL GRADE (100%) is the sum of FINAL EXAM+LAB 1+LAB 2+LAB 3+PROJECT with a 'minimum score' condition set on the item FINAL EXAM.
        - FINAL EXAM: 40% (minimun score set on 50 out of 100)
        - LAB 1: 10%
        - LAB 2: 15%
        - LAB 3: 15%
        - PROJECT: 20%

The instructor might request their students that in order to pass the course they need to get at least a 50% in the score of the final exam (this is why I said in my first comment that this might look like a conditional release).
Then, a student could have top grades in LAB 1, LAB 2, LAB 3 and PROJECT (60% of the final grade) but do bad on the FINAL EXAM and recieve just a 30% out of 100% which in is a 12% out the 40% that the FINAL EXAM is worth within the final grade. Since the 'minimun score' for the gradebook item FINAL EXAM was set to 50% and the student got a 30%, then, the final grade wont be calculated. That student will need to take that FINAL EXAM again until he gets the 50% requested by the instructor in order to pass the course.

Does it make more sense now? As I said, it is a real use case from many instructors at our institution. Right now, what they do, is that they handle the minimun score in a manual way, if the student has not reached that minimun score in a gradebook item, they don't add the FINAL GRADE to the university SIS (Student Information System) and they ask the student to re-do/re-take the gradebook item that they are missing.

**Comment by steveswinsburg** [comment]
To clarify just a bit more, if the minimum grade isn't met for a gradebook item, is that item excluded from the course grade calculations (so if you had 5 items and only 4 qualified, only the points from the 4 would be used) or does the course grade just not get calculated at all unless all 5 qualify?

Seems pretty simple to add a flag to each item which then contributes to the calculations, just need to make sure the priorities and community requirements are managed.

**Comment by jonespm** [comment]
Yeah, I think that's most of my problem is the ambiguity of what "Minimum Score" would do. There's already "Drop lowest" but this might not be the lowest and it doesn't sound like he's saying he wants it dropped or not included in the calculations, he's saying he doesn't want any calculation at all. I've never heard of a grading system like this and I think it would be something you'd want to "turn off" or hide if it was implemented because it would be confusing.

In any case I'd probably expand the description from "Minimum Score" to "Minimum item score required for final grade calculation" even if it is wordy.
